internal/pkg/template/workload.go

```diff
@@ -291,14 +280,39 @@ func (cfg *NetworkLoadBalancer) Aliases() []string {
 	for _, listener := range cfg.Listener {
 		for _, entry := range listener.Aliases {
 			if _, value := seen[entry]; !value {
+				uniqueAliases = append(uniqueAliases, entry)
 				seen[entry] = true
+			}
+		}
+	}
+	return uniqueAliases
+}
+
+// Aliases return all the unique aliases specified across all the routing rules in ALB.
+// Currently, we only have primary routing rule, but we will be getting additional routing rule soon.
+func (cfg *ALBListener) Aliases() []string {
+	var uniqueAliases []string
+	seen := make(map[string]bool)
+	for _, rule := range cfg.Rules {
+		for _, entry := range rule.Aliases {
+			if _, value := seen[entry]; !value {
 				uniqueAliases = append(uniqueAliases, entry)
+				seen[entry] = true
 			}
 		}
 	}
 	return uniqueAliases
```

Comment:
nit:

There are two pros of using `map[string]struct{}` instead of `map[string]bool`:
1. A `struct{}{}` takes zero byte
2. It is ambiguous what `seen["something"] = false` means

```suggestion
	var uniqueAliases []string
	seen := make(map[string]struct{}{})
	exists := struct{}{}
	for _, rule := range cfg.Rules {
		for _, entry := range rule.Aliases {
			if _, ok := seen[entry]; !ok {
				uniqueAliases = append(uniqueAliases, entry)
				seen[entry] = exists
			}
		}
	}
	return uniqueAliases
```